aten/src/ATen/native/cuda/ForeachPointwiseOp.cu

```diff
@@ -74,7 +76,40 @@ void foreach_tensor_##NAME##_cuda_(TensorList input, TensorList tensors1, Tensor
     foreach_pointwise_op_<OP>(input, tensors1, tensors2, scalar);                                                              \
 }
 
-FOREACH_UNARY_OP(addcmul, std::multiplies);
-FOREACH_UNARY_OP(addcdiv, std::divides);
+FOREACH_POINTWISE_OP(addcmul, std::multiplies);
+FOREACH_POINTWISE_OP(addcdiv, std::divides);
+
+std::vector<Tensor> foreach_tensor_max_cuda(TensorList tensors1, TensorList tensors2) {
+    TORCH_CHECK(tensors1.size() > 0, "Tensor list must have at least one tensor.");
+    TORCH_CHECK(tensors1.size() ==  tensors2.size(), "Tensor lists must be of the same length.");
+
+    if (!can_use_fast_route(tensors1, tensors2)) {
+        return at::native::foreach_tensor_max_slow(tensors1, tensors2);
+    }
+
+    std::vector<std::vector<at::Tensor>> tensor_lists;
+    std::vector<at::Tensor> vec_res;
+    vec_res.reserve(tensors1.size());
+    for (const auto& t: tensors1) {
+        vec_res.emplace_back(at::native::empty_like(t));
+    }
+
+    tensor_lists.emplace_back(tensors1.vec());
+    tensor_lists.emplace_back(tensors2.vec());
+    tensor_lists.emplace_back(std::move(vec_res));
+    auto res = std::max<int> (2, 3); // WORKS
+    auto res2 = std::max<float> (2, 3); // WORKS
+    
+    
+    AT_DISPATCH_ALL_TYPES_AND(kHalf, tensors1[0].scalar_type(), "foreach_pointwise_op_cuda", [&]() {
+        using opmath_t = get_opmath_t<scalar_t>::opmath_t;
+        auto op = []  __device__ (opmath_t a, opmath_t b) -> opmath_t { return std::max(a, b); };
```

Comment:
Maybe replace `__device__` with GPU_LAMBDA?

Reply:
Done. the error is still there -` terminated by signal SIGSEGV (Address boundary error)`